pandas/tests/io/test_html.py

```diff
@@ -373,32 +373,6 @@ def test_python_docs_table(self):
         zz = [df.iloc[0, 0][0:4] for df in dfs]
         assert sorted(zz) == sorted(["Repo", "What"])
 
-    @pytest.mark.slow
-    def test_thousands_macau_stats(self, datapath):
-        all_non_nan_table_index = -2
-        macau_data = datapath("io", "data", "html", "macau.html")
-        dfs = self.read_html(macau_data, index_col=0, attrs={"class": "style1"})
-        df = dfs[all_non_nan_table_index]
-
-        assert not any(s.isna().any() for _, s in df.items())
-
-    @pytest.mark.slow
-    def test_thousands_macau_index_col(self, datapath, request):
```

Comment:
Was there no replacement for this test? It highlight behaviour we where hoping to fix in https://github.com/pandas-dev/pandas/issues/29622 ?

Reply:
I seem to have misinterpreted this issue as "blank _column names_ should not load as NaN", after observing that empty headers loaded as "Unnamed: n" while empty dtype=str body cells loaded as NaN.  (I hence added "assert "Unnamed" in result.columns[-1]" to check for that.)

I now suspect this was because I was testing this in bs4 4.8.

wikipedia_states has empty body cells, so re-adding this check should be easy.  (Possibly relevant to the "what caused this" discussion: it also has nested tables but they aren't the one being read.)